File: frontend/app/graphql/fragments/Grootboekrekening.ts
import {gql} from "@apollo/client";

export const GrootboekrekeningFragment = gql`
    fragment Grootboekrekening on Grootboekrekening {
        id
        naam
        credit
        omschrijving
        referentie
        rubriek{
            id
            naam
        }
    }
`;

File: backend/hhb_backend/service/model/grootboekrekening.py
from typing import Optional, List

from hhb_backend.service.model.base_model import BaseModel


class Grootboekrekening(BaseModel):
    id: str
    naam: str
    referentie: str
    omschrijving: str
    credit: bool
    parent_id: Optional[int]
    # exposed relations
    children: List[str]
